### PR TITLE
Refactor InternalLogger class

### PR DESCRIPTION
**Pull request by ie-zero**
Refactor InternalLogger class to improve code readabiity and create further opportunities to improve the code.

There are no functional changes or changes to the public interface.

The method Write() in InternalLogger class split into smaller methods
* FormatMessage()
* WriteToLogFile()
* WriteToTextWriter()
* WriteToConsole()
* WriteToErrorConsole()

LoggingEnabled() method renamed to IsLoggingEnabled() in alignment to IsSeriousException(). It also convey better to the reader the purpose of the method.

### CODE REVIEW

File: src/NLog/Common/InternalLogger.cs [hunk 1]
```diff
@@ -336,6 +280,39 @@ private static void Write([CanBeNull]Exception ex, LogLevel level, string messag
...
 
+        private static string FormatMessage([CanBeNull]Exception ex, LogLevel level, string message, [CanBeNull]object[] args)
+        {
+            const string TimeStampFormat = "yyyy-MM-dd HH:mm:ss.ffff";
+            const string FieldSeparator = " ";
+
+            var formattedMessage =
+                (args == null) ? message : string.Format(CultureInfo.InvariantCulture, message, args);
+
+            var builder = new StringBuilder(formattedMessage.Length + TimeStampFormat.Length + 8);
```
Comment (304NotModified) [inline src/NLog/Common/InternalLogger.cs]: Also small change here I see, maybe also check `IncludeTimestamp`?

File: src/NLog/Common/InternalLogger.cs [hunk 2]
```diff
@@ -367,6 +344,64 @@ private static bool LoggingEnabled(LogLevel logLevel)
                    LogWriter != null;
         }
 
+        private static void WriteToLogFile(string message)
```
Comment (304NotModified) [inline src/NLog/Common/InternalLogger.cs]: Please add some docs, most important is here that logging will be skipped if not enabled. (multiple times)

Comment (ie-zero) [reply src/NLog/Common/InternalLogger.cs]: I will do

File: src/NLog/Common/InternalLogger.cs [hunk 3]
```diff
@@ -288,7 +288,7 @@ private static string FormatMessage([CanBeNull]Exception ex, LogLevel level, str
             var formattedMessage =
                 (args == null) ? message : string.Format(CultureInfo.InvariantCulture, message, args);
 
-            var builder = new StringBuilder(formattedMessage.Length + TimeStampFormat.Length + 8);
+            var builder = new StringBuilder(formattedMessage.Length + TimeStampFormat.Length + ex?.ToString()?.Length ?? 0 + 25);
```
Comment (304NotModified) [inline src/NLog/Common/InternalLogger.cs]: 0 + 25? Should be 8+25?

or 0 + 0+ 0+ 0+ 0+ 0+ 0+ 0 + 25 :P

Comment (ie-zero) [reply src/NLog/Common/InternalLogger.cs]: ```formattedMessage.Length``` and ```TimeStampFormat.Length``` (constant length) should never be 0. ```ex?.ToString()?.Length ?? 0``` might be zero and the rest of white-spaces the literal 'Exception' and the LogLevel as a string sum up, if I recall correctly, to 23 characters.

To be honest, I am not certain if pre-calculating the length of the expected string in such detail for optimisation is positive. 

What do you think?

Comment (304NotModified) [reply src/NLog/Common/InternalLogger.cs]: true, pre-optimized could be done later. 

`0 + 25` looks just odd. Its 25 then.

File: src/NLog/Common/InternalLogger.cs [hunk 4]
```diff
@@ -336,6 +280,39 @@ private static void Write([CanBeNull]Exception ex, LogLevel level, string messag
...
 
+        private static string FormatMessage([CanBeNull]Exception ex, LogLevel level, string message, [CanBeNull]object[] args)
+        {
+            const string TimeStampFormat = "yyyy-MM-dd HH:mm:ss.ffff";
+            const string FieldSeparator = " ";
+
+            var formattedMessage =
+                (args == null) ? message : string.Format(CultureInfo.InvariantCulture, message, args);
+
+            var builder = new StringBuilder(formattedMessage.Length + TimeStampFormat.Length + ex?.ToString()?.Length ?? 0 + 25);
```
Comment (304NotModified) [inline src/NLog/Common/InternalLogger.cs]: Resharper says this is  ` ex?.ToString()?.Length ?? (0 + 25)`

So this is wrong ;)

**Comment by 304NotModified** [comment]
Merged. Thanks!

**Pull request merged**